lesson1/train.rb

```diff
@@ -0,0 +1,68 @@
+class Train
+  attr_reader :amount, :number, :type, :speed
+
+  def initialize(number, type, amount)
+    @number = number
+    @type = type
+    @amount = amount
+    @speed = 0
+    @route = nil
+    @current_station = nil
+  end
+
+  def increase_speed(value)
+    @speed += value
+  end
+
+  def decrease_speed(value)
+    @speed - value < 0 ? @speed = 0 : @speed -= value 
+  end
+
+  def add_train_car
+    @amount += 1
+  end
+
+  def remove_train_car
+    if @speed == 0
+      @amount == 0 ? @amount = 0 : @amount -= 1
+    end
+  end
+
+  def route=(route)
+    @route = route
+    @route.stations.first.add_train(self)
+    @current_station = 0
+  end
+
+  def move_forward
+    if @current_station < @route.stations.length - 1
+      @route.stations[@current_station].remove_train(self)
+      @current_station += 1
+      @route.stations[@current_station].add_train(self)
+    end
+  end
+
+  def move_backward
+    if @current_station > 0
+      @route.stations[@current_station].remove_train(self)
+      @current_station -= 1
+      @route.stations[@current_station].add_train(self)
+    end
+  end
+
+  def get_prev_station
+    if @current_station > 0
```

Comment:
в первом ревью был такой комментарий: https://github.com/Simonion19/RoR/pull/1#discussion_r395184106
так вот если бы в классе Роут были методы "первая станция" , "последняя станция", как там рекомендуется, то здесь можно бы было так написать:
```suggestion
    if current_station != @route.first_station
```